harvard-lts/CURIOSity
# Language options now appear for anonymous users.

**Language options now appear for anonymous users.**
* * *

**JIRA Ticket**: [LTSMAINT-755](https://jira.huit.harvard.edu/browse/LTSMAINT-755)

# What does this Pull Request do?
This displays the navigation menu to toggle the languages if the exhibit has translations. An anonymous user would see the language to the right of the Harvard Library logo, as this is Spotlight's out-of-the-box location for this. For example:

 
![Italiano-Local](https://github.com/harvard-lts/CURIOSity/assets/40801910/5a18b5b4-be1b-4fa3-be57-ef02ac959d4e)

Note that I modified `app/views/shared/_header_navbar.html.erb` so that empty markup does not appear if the exhibit does not have additional languages / translations. The if statement was taken from Spotlight core here, so it should be the valid / accepted way to check for this: https://github.com/projectblacklight/spotlight/blob/main/app/views/spotlight/shared/_locale_picker.html.erb#L1

# How should this be tested?

A description of what steps someone could take to:
* Rebuild curiosity off of the `LTSMAINT-755` branch
* Login as an administrator and edit one of the exhibits (I used the "Demo" exhibit because this is already set up on Production).
* Under the "General" tab click on "Languages"
* Select Language "Italian" and press the "Create Language" button
* Under "Current Languages" Next to "Italian" click the "Public" checkbox and press the "Save Changes" button
* **IN CURIOSITY PRODUCTION:** login as an admin and go to the Demo Exhibit Dashboard. Click on the "Translations" link in the left navigation. Click the "Export Italian" button.
* **BACK IN YOUR LOCAL SITE**: click on the "Translations" link in the left navigation, click on the "Browse" button in the "Import a YAML file" text box and upload the YML file you downloaded from the previous step.
* Confirm that you can see the "Italiano" button at the top of every page on the "Demo" exhibit and can toggle between Italian and English when you are both logged in and logged out.
* Confirm that nothing else shows up in that space when you are logged out and browsing other exhibits.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@enriquediaz @dl-maura @mferrarini
